test/integration/job/job_test.go

```diff
@@ -4095,6 +4095,83 @@ func TestSuspendJob(t *testing.T) {
 	}
 }
 
+// TestSuspendJobStartTimeValidation verifies that the job controller can update startTime
+// when resuming a suspended job (https://github.com/kubernetes/kubernetes/issues/134521).
+func TestSuspendJobStartTimeValidation(t *testing.T) {
```

Comment:
I would also suggest to enable the JobManagedBy feature gate explicitly, even if now it is enabled by default.